File: JavaScript/60. Permutation Sequence.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function(n, k) {
    var res = "";
    var f=1;
    var arr = new Array();
    for(var i = 1; i<n; i++){
        f*=i;
        arr.push(i);
    }
    arr.push(n);
    k--;
    while(arr.length>0){
        res+=arr[Math.floor(k/f)];
        arr.splice(Math.floor(k/f),1);
        k%=f;
        if(arr.length!=0){
            f=Math.floor(f/arr.length);
        }
    }
    return res;
};

File: Java/60. Permutation Sequence.java
class Solution {
    public String getPermutation(int n, int k) {
        String res = "";
        int f=1;
        List<Integer> l = new ArrayList<>();
        for(int i = 1; i<n; i++){
            f*=i;
            l.add(i);
        }
        l.add(n);
        k--;
        while(l.size()>0){
            res+=l.get(k/f);
            l.remove(k/f);
            k%=f;
            if(l.size()!=0){
                f/=l.size();
            }
        }
        return res;
    }
}
